src/widgets/widget-model.js

```diff
@@ -72,8 +72,14 @@ module.exports = cdb.core.Model.extend({
 
     var styles = this.get('style');
 
-    if ((!styles || !styles.auto_style) && (this.get('type') === 'category' || this.get('type') === 'histogram')) return true;
-    return styles && styles.auto_style && styles.auto_style.allowed;
+    if (this.get('type') === 'category' || this.get('type') === 'histogram') {
+      if (!styles || !styles.auto_style) {
```

Comment:
not sure if we can short this as `return styles && styles.auto_style && styles.auto_style.allowed || true`